doc/specifications/proposed/named_queries/developer_documentation.md

```diff
@@ -0,0 +1,181 @@
+
+#### Content Query
+The Query results are available as `content_array`, an array of ContentInfo objects.
+
+## QueryType objects
+To make a new QueryType available to the Query Controller, you need to register a new
+QueryType and register it.
+
+A QueryType is a PHP class that implements QueryType interface (`eZ\Publish\Core\MVC\Query\QueryType`).
+It has two methods:
+- `getName(): string`
+  Returns the QueryType name.
+- `getQuery( array $parameters = null ): Query`
+  It accepts an optional hash of parameters that can be used in any way in the implementation:
+  - customize an element's value (limit, ContentType identifier, etc)
+  - conditionally add/remove criteria from the query
+  - set the limit/offset...
+  It returns a Query object, that can be either a LocationQuery or a (Content) Query.
+
+
+### QueryType example: latest content
+This QueryType returns a LocationQuery that searches for the 10 last published content, order by reverse
+publishing date. It accepts an optional `type` parameter, that can be set to a ContentType identifier.
+identifier:
+
+```php
+<?php
+namespace AcmeBundle\eZ\Query;
+
+use eZ\Publish\Core\MVC\Query\QueryType;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+
+class LatestContentQueryType implements QueryType
+{
+  public function getQuery(array $parameters = [])
+  {
+    $query = new LocationQuery();
+    $query->criterion = new Criterion\LogicalAnd([
+      $criteria[] = new Criterion\Visibility(Criterion\Visibility::VISIBLE);
+      if (isset($parameters['type'])) $criteria[] = new Criterion\ContentTypeIdentifier('article');
+      $criteria[] = new Criterion\ParentLocationId(2);
+      $criteria[] = new Criterion\Location\Depth(Criterion\Operator:GT, 1)
+    ]);
+    $query->sortClauses = [new SortClause\DatePublished()];
+    $query->limit = 10;
+
+    return $query;
+  }
+
+  public function getName()
+  {
+    return 'LatestContent';
+  }
+}
+```
+
+### Registering a QueryType
+In addition to creating a PHP class for a `QueryType`, it must also be registered
+within the Service Container. This can be done in two ways: by convention, and
+with a service tag.
+
+#### By convention
+Any class named `AcmeBundle\eZ\QueryType\*QueryType`, that implements
+the QueryType interface, will be registered as a
+custom QueryType.
+
+#### Using a service tag
+```yaml
+acme.query.latest_content:
+    class: AcmeBundle\Query\LatestContent
+    tags:
+        - {name: ezpublish.query_type}
+```
+
+## Using QueryTypes from PHP code
+All QueryTypes are registered in a registry, the QueryType registry.
+
+It is available from the container as `ezpublish.query_type.registry`:
+
+```php
+class MyController extends Controller
+{
+    /** @var QueryTypeRegistry */
+    private $queryTypeRegistry;
+
+    public function myAction()
+    {
+        $queryType = $this->queryTypeRegistry->getQueryType( 'AcmeBundle:LatestContent' );
+        $query = $queryType->getQuery(['type' => 'article']);
```

Comment:
i'd expand this to add the render part and show how you can pass location_list or content_list to the template...